commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/commands/updateactions/AddLineItem.java

```diff
@@ -0,0 +1,80 @@
+package io.sphere.sdk.shoppinglists.commands.updateactions;
+
+import io.sphere.sdk.commands.UpdateActionImpl;
+import io.sphere.sdk.products.ProductIdentifiable;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import io.sphere.sdk.types.CustomDraft;
+import io.sphere.sdk.types.CustomFieldsDraft;
+
+import javax.annotation.Nullable;
+import java.time.ZonedDateTime;
+
+/**
+ * Adds a line item.
+ *
+ * {@doc.gen intro}
+ *
+ * {@include.example io.sphere.sdk.shoppinglists.commands.ShoppingListUpdateCommandIntegrationTest#addLineItem()}
+ *
+ * @see ShoppingList#getLineItems()
+ */
+public final class AddLineItem extends UpdateActionImpl<ShoppingList> implements CustomDraft {
+    private final String productId;
+    @Nullable
+    private final Integer variantId;
+    @Nullable
+    private final Long quantity;
+    @Nullable
+    private final ZonedDateTime addedAt;
+    @Nullable
+    private final CustomFieldsDraft custom;
+
+    private AddLineItem(final String productId, final Integer variantId, @Nullable final Long quantity, @Nullable final ZonedDateTime addedAt, @Nullable final CustomFieldsDraft custom) {
+        super("addLineItem");
+        this.productId = productId;
+        this.variantId = variantId;
+        this.quantity = quantity;
+        this.addedAt = addedAt;
+        this.custom = custom;
+    }
+
+    public static AddLineItem of(final ProductIdentifiable product, final int variantId, final Long quantity) {
+        return of(product.getId(), variantId, quantity);
+    }
+
+    public static AddLineItem of(final String productId, final int variantId, final Long quantity) {
```

Comment:
Oh and variant too. I would keep a constructor with product, variant and quantity together, for keeping the similarity with cart's line item, but I would provide one constructor with only product and enable `with` methods for the rest.